include/drivers/mm/mm_drv_intel_adsp_mtl_tlb.h

```diff
@@ -1,21 +1,40 @@
-/* SPDX-License-Identifier: Apache-2.0 */
-/*
- * Copyright (c) 2022 Intel Corporation.
- *
- * Author: Marcin Szkudlinski <marcin.szkudlinski@linux.intel.com>
- *
- */
-
-#ifndef ZEPHYR_INCLUDE_DRIVERS_INTEL_ADSP_MTL_TLB
-#define ZEPHYR_INCLUDE_DRIVERS_INTEL_ADSP_MTL_TLB
-
-/* device id from the device tree */
-#define DT_DRV_COMPAT_ADSP_TLB intel_adsp_tlb
-
-/* size of TLB table */
-#define TLB_SIZE DT_REG_SIZE_BY_IDX(DT_INST(0, DT_DRV_COMPAT_ADSP_TLB), 0)
-
-/* numer of L2 memory pages */
-#define L2_SRAM_PAGES_NUM (L2_SRAM_SIZE / CONFIG_MM_DRV_PAGE_SIZE)
-
-#endif /* ZEPHYR_INCLUDE_DRIVERS_INTEL_ADSP_MTL_TLB */
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Author: Marcin Szkudlinski <marcin.szkudlinski@linux.intel.com>
+ *
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_INTEL_ADSP_MTL_TLB
+#define ZEPHYR_INCLUDE_DRIVERS_INTEL_ADSP_MTL_TLB
+
+/* device id from the device tree */
+#define DT_DRV_COMPAT_ADSP_TLB intel_adsp_tlb
+
+/* size of TLB table */
+#define TLB_SIZE DT_REG_SIZE_BY_IDX(DT_INST(0, DT_DRV_COMPAT_ADSP_TLB), 0)
+
+/* numer of L2 memory pages */
+#define L2_SRAM_PAGES_NUM (L2_SRAM_SIZE / CONFIG_MM_DRV_PAGE_SIZE)
+
+/* maximum size of buffer required for memory save */
+#define TLB_MAX_MEMORY_SAVE_BUFFER_SIZE \
+	(L2_SRAM_SIZE + TLB_SIZE + L2_SRAM_PAGES_NUM * sizeof(void *))
+
```

Comment:
I wonder if all those defines should actually belong to an "internal" `drivers/mm/mm_drv_intel_adsp_mtl_tlb.h` and only the API bits here.